tasks/build/os_shell_scripts.js

```diff
@@ -0,0 +1,43 @@
+import {unlink} from 'fs';
+import {join, extname} from 'path';
+import {promisify} from 'bluebird';
+import {ncp} from 'ncp';
+const pncp = promisify(ncp);
+const punlink = promisify(unlink);
+
+export default function (grunt) {
+  grunt.registerTask('_build:osShellScripts', function () {
+    const done = this.async();
+    const source = 'build/kibana/bin';
+    const platforms = grunt.config.get('platforms');
+    const allPlatforms = fn => invokeAllAsync(platforms, fn);
+
+    run()
+      .then(() => done())
+      .catch(e => done(e));
+
+    async function run() {
```

Comment:
I don't think you need to wrap this in a new function, you can just make the overall task function async:

```js
grunt.registerTask('_build:osShellScripts', async function () {
  const done = this.async();
  // ...
  try {
    await allPlatforms(platform => punlink(join(platform.buildDir, 'bin')));
    await allPlatforms(platform => pncp(source, join(platform.buildDir, 'bin')));
    await allPlatforms(removeExtraneousShellScripts);
    done();
  } catch (e) {
    done(e);
  }
});
```